specs/Standards/StorageManagement.md

```diff
@@ -0,0 +1,423 @@
+
+**High-level explanation**
+
+1. Alice issues a transaction, let's say `social.post`, and it fails with an intelligible error message to tell her that she has an insufficient storage balance to cover the cost of the operation
+2. Alice issues a transaction to increase her storage balance
+3. Alice retries the initial transaction and it succeeds
+
+**Technical calls**
+
+1. This is outside the scope of this spec, but let's say Alice calls `near call social post '{ "text": "very long message" }'`, and that this fails with a message saying something like "Unsufficient storage deposit for transaction. Please call `storage_deposit` and attach at least 0.1 NEAR, then try again."
+
+2. Alice deposits the proper amount in a transaction by calling `social::storage_deposit` with the attached deposit of '0.1'. Using NEAR CLI:
+
+       near call social storage_deposit '' \
+         --accountId alice --amount 0.1
+
+   The result:
+
+       {
+         total: '200000000000000000000000',
+         available: '100100000000000000000000'
+       }
+
+3. Alice tries the initial `near call social post` call again. It works.
+
+#### 4. Removing storage and reclaiming excess deposit
+
+Assumption: Alice has more deposited than she is using.
+
+**High-level explanation**
+
+1. Alice views her storage balance and sees that she has extra.
+2. Alice withdraws her excess deposit.
+
+**Technical calls**
+
+1. Alice queries `social::storage_balance_of({ "account_id": "alice" })`. With NEAR CLI:
+
+       near view social storage_balance_of '{"account_id": "alice"}'
+
+   Response:
+
+       {
+         total: '200000000000000000000000',
+         available: '100100000000000000000000'
+       }
+
+2. Alice calls `storage_withdraw` with a 1 yoctoⓃ deposit. NEAR CLI command:
+
+       near call social storage_withdraw \
+         '{"amount": "100100000000000000000000"}' \
+         --accountId alice --amount .000000000000000000000001
+
+   Result:
+
+       {
+         total: '200000000000000000000000',
+         available: '0'
+       }
+
+## Reference-level explanation
+
+**NOTES**:
+
+- All amounts, balances and allowance are limited by `U128` (max value 2<sup>128</sup> - 1).
+- This storage standard uses JSON for serialization of arguments and results.
+- Amounts in arguments and results are serialized as Base-10 strings, e.g. `"100"`. This is done to avoid JSON limitation of max integer value of 2<sup>53</sup>.
+- To prevent the deployed contract from being modified or deleted, it should not have any access keys on its account.
+
+**Interface**:
+
+```ts
+// The below defines the structure that will be returned for the methods:
+// * `storage_deposit`
+// * `storage_withdraw`
+// * `storage_balance_of`
+// The `total` and `available` values are string representations of unsigned
+// 128-bit integers.
+type StorageBalance = {
+   total: string;
+   available: string;
+}
```

Comment:
Need better explanation what is `total` and `available`. Mention that they are the balance of tokens in yoctoNEAR.

Reply:
Updated to  

```
// The `total` and `available` values are string representations of unsigned
// 128-bit integers showing the balance of a specific account in yocto
```